src/dataset/composer.py

```diff
@@ -34,3 +34,45 @@ def __call__(self, image: np.ndarray, annotations: Dict) -> Any:
             image, annotations = func(image, annotations, **self.kwargs)
 
         return image, annotations
+
+
+def join_patches(patches: List[np.ndarray], filenames: List[str]) -> np.ndarray:
+    if len(patches) == 0:
+        raise ValueError("Input list of patches is empty.")
+
+    patches_positions = [x_y_from_filename(filename) for filename in filenames]
+    result_shape = calculate_result_shape_from_patches(patches, patches_positions)
+    result_image = np.zeros(result_shape, dtype=np.uint8)
+
+    for (y_at_result_image, x_at_result_image), patch in zip(patches_positions, patches):
+        height, width = patch.shape
+
+        result_image[
+            y_at_result_image:y_at_result_image + height,
+            x_at_result_image:x_at_result_image + width
+        ] = np.maximum(
+            result_image[y_at_result_image:y_at_result_image + height, x_at_result_image:x_at_result_image + width],
+            patch)
+
+    return result_image
+
+
+def calculate_result_shape_from_patches(patches: List[np.ndarray], dimensions: Tuple[int, int]) -> Tuple[int, int]:
+    max_x = 0
+    max_y = 0
+    for (x, y), patch in zip(dimensions, patches):
+        patch_height, patch_width = patch.shape
+        if x + patch_width > max_x:
+            max_x = x + patch_width
+        if y + patch_height > max_y:
+            max_y = y + patch_height
+
+    return (max_x, max_y)
```

Comment:
A more pythonic alternative would be something like:

```python
height, width = patches[0].shape[:2]

xs, ys = list(zip(*dimensions))   # [(x1, y1), (x2, y2), ..., (xn, yn)] -> ((x1, x2, ..., xn), (y1, y2, ..., yn))
max_x, max_y = max(xs) + width, max(ys) + height
```
In the snippet above, I considered patches as having the same shape (they normally do and it is ok to assume that). But if you want to make it robust against patches of multiple dimensions then use arrays:

```python
coords = list(zip(*dimensions))
shapes = list(zip(*[patch.shape[:2] for patch in patches]))

image_shape = np.max(np.array(shapes) + np.array(coords), axis=1)
```